File: BackEnd/src/main/java/com/example/wisdombooks/entity/Favorite.java
package com.example.wisdombooks.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

@Data
@TableName("favorite_list")
public class Favorite {
    // 身份证号
    private String idCard;
    // 书籍编号
    private String bNo;
    // 书名
    private String bookName;
}

File: FontEnd/src/utils/api/favorite.js
import { favorite } from "./http.js";

const { get, put } = favorite;

const favoriteBook = async (bNo, bookName) => {
  try {
    await put("", {
      bNo,
      bookName,
    });
  } catch (e) {
    return false;
  }
  return true;
};

const unFavoriteBook = async (bNo, bookName) => {
  try {
    await favorite.delete("", {
      data: {
        bNo,
        bookName,
      },
    });
  } catch (e) {
    return false;
  }
  return true;
};

const getFavoriteList = async () => {
  const data = await get("");
  return data;
};

const getBookInfo = async (bNo) => {
  const data = await get("to", {
    params: {
      bNo,
    },
  });
  return data;
};

export { favoriteBook, getBookInfo, getFavoriteList, unFavoriteBook };
